sys/kern/vfs_aio.c

```diff
@@ -1586,7 +1599,25 @@ aio_aqueue(struct thread *td, struct aiocb *ujob, struct aioliojob *lj,
 	kev.filter = EVFILT_AIO;
 	kev.flags = EV_ADD | EV_ENABLE | EV_FLAG1 | evflags;
 	kev.data = (intptr_t)job;
-	kev.udata = job->uaiocb.aio_sigevent.sigev_value.sival_ptr;
+#ifdef COMPAT_CHERIABI
+	if (SV_CURPROC_FLAG(SV_CHERI)) {
+		/*
+		 * In a CheriABI process, we use sidecar storage for
+		 * both kqueue and sigevent pointers.  The two merge
+		 * here!
+		 *
+		 * XXXBD: There has got to be a better way to do this...
+		 */
```

Comment:
I don't see how this works, or is this the "normal" thing done for other CheriABI kevents?  (I don't see anything in the diff that unpacks the sidecar at least.)  Perhaps there should be a 'cheriabi_build_kevent_udata(kev, ident, udata)' helper function if so so that the logic is centralized in kern_event.c?

Reply:
It's handled in the kevent code.  I've added more abstraction.